### ISSUE TITLE
isolatedDeclarations does not flag potential type predicate funcs

### ISSUE THREAD

**Issue by jakebailey** [microsoft/typescript-go#3193]
```ts
// @isolatedDeclarations: true
// @declaration: true

export function isString(value: unknown) {
  return typeof value === "string";
}
```

This should produce an error, but does not. We seemingly do not have any tests for this.
